src/miral/basic_window_manager.cpp

```diff
@@ -535,15 +535,27 @@ void miral::BasicWindowManager::focus_next_application()
             }
             while (focus_controller->focused_session() != prev.application());
         }
+        else
```

Comment:
I would swap the order of this if/else block. Just a personal preference one doesn't inverse the if condition if there is not a clear order.

(What I mean is
```
if (foo.empty())
{...}
else
{...}
```
Is slightly more readable then
```
if (!foo.empty())
{...}
else
{...}
```
)